### PR TITLE
Keycloak Addon Support

### PR DESCRIPTION
**Pull request by voatsap**
As Kubernetes Terraform module addons:

* Deploy Operator Lifecycle Manager
* Deploy Keycloak Operator
* Deploy Keycloak
* Output Keycloak domain & credentials

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -6,7 +6,9 @@ outputs:
     description: 'An status for the job'
 runs:
   using: 'docker'
-  image: 'docker://clusterdev/cluster.dev:0.1.10'
+  #image: 'docker://clusterdev/cluster.dev:0.1.10'
```
Comment (voatsap) [inline action.yml:9]: would be updated before release

File: terraform/aws/addons/olm.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+resource "null_resource" "olm_install" {
+  provisioner "local-exec" {
+    # Deploy CRD's, wait them become available and deploy olm
+    command = "kubectl apply -f ${local.url}/${local.olm-version}/crds.yaml && kubectl apply -f ${local.url}/${local.olm-version}/olm.yaml && for resource in catalog-operator olm-operator packageserver; do kubectl rollout status -n ${local.olm-namespace} -w deploy $resource; done"
+  }
+  provisioner "local-exec" {
+    when = destroy
+    # No sense to delete it
+    command = "exit 0"
+  }
```
Comment (MaxymVlasov) [inline terraform/aws/addons/olm.tf:17]: [p2] How user can disable/uninstall only OLM without cluster recreation?

File: terraform/aws/addons/keycloak.tf [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Deploy keycloak-operator and keycloak itself
+locals {
+  cluster_issuer     = "letsencrypt-prod"
```
Comment (MaxymVlasov) [inline terraform/aws/addons/keycloak.tf:3]: [p4] Why suffix `-prod` exist?

File: terraform/aws/addons/templates/keycloak-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
```
Comment (MaxymVlasov) [inline terraform/aws/addons/templates/keycloak-operator.yaml:1]: [p5] Good to have separator on top of file for hypothetical case when YAML's will be merged.
```suggestion
---
apiVersion: v1
```

File: terraform/aws/addons/templates/keycloak.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+apiVersion: keycloak.org/v1alpha1
```
Comment (MaxymVlasov) [inline terraform/aws/addons/templates/keycloak.yaml:1]: [p5] Same
```suggestion
---
apiVersion: keycloak.org/v1alpha1
```

File: terraform/aws/addons/templates/keycloak.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+kind: Keycloak
+metadata:
+  name: keycloak
+  namespace: ${namespace}
+  labels:
+    app: keycloak
+spec:
+  instances: 1
+  extensions:
+    - https://github.com/aerogear/keycloak-metrics-spi/releases/download/1.0.4/keycloak-metrics-spi-1.0.4.jar
```
Comment (MaxymVlasov) [inline terraform/aws/addons/templates/keycloak.yaml:11]: [p4] Maybe good to manage extension version from terraform

File: terraform/aws/addons/olm.tf (hunk 2 above)
Comment (voatsap) [reply terraform/aws/addons/olm.tf:17]: Not possible right now due to hanging delete olm when resources provisionet with it.

File: terraform/aws/addons/keycloak.tf (hunk 3 above)
Comment (voatsap) [reply terraform/aws/addons/keycloak.tf:3]: Its common name for letsencrypt provisioners: letsencrypt-stage - for testing purpose and prod - for normal usage

File: terraform/aws/addons/templates/keycloak.yaml (hunk 6 above)
Comment (voatsap) [reply terraform/aws/addons/templates/keycloak.yaml:11]: passed template as is. not in scope

**Pull request merged**